app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/EditGuestAccessViewModel.kt

```diff
@@ -162,6 +162,9 @@ class EditGuestAccessViewModel @Inject constructor(
 
         viewModelScope.launch {
             withContext(dispatcher.io()) {
+                if (!shouldEnableGuestAccess) {
+                    removeGuestLink()
```

Comment:
Then switching the guests allowed switch, will trigger the display of the revoke guest link?? Would that be ok from product perspective or a bit confusing?